spec/datadog/appsec/utils/read_write_lock_spec.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+require 'datadog/appsec/utils/read_write_lock'
+
+RSpec.describe Datadog::AppSec::Utils::ReadWriteLock do
+  subject(:read_write_lock) { described_class.new }
+
+  context 'read locks' do
+    it 'allows for many readers' do
+      rw_lock = read_write_lock
+      state = 0
+      read_threads = Array.new(3) do
+        Thread.new do
+          rw_lock.with_rlock do
+            expect(state).to eq(0)
+          end
+        end
+      end
```

Comment:
This expectation is a bit weak, since it's trivially realized even if the code never runs.

One possible idea to make this test a bit stronger is perhaps having each thread have an id (e.g. `(1..3).map` ...) and then each thread could put is own id on a `Queue`. At the end of the test, you should be able to pop all 3 distinct ids out of the queue.